### DISCUSSION TITLE
ChromeOS stopped booting - can I recover my personal files?

### DISCUSSION THREAD

**Discussion by Vlad-BE** [sebanc/brunch#1687] [General]
Hi All, unfortunately today ChromeOS stopped booting on my Lenovo X260 all of a sudden. I only see Lenovo's logo going on and off in a loop. I have been using ChromeOS for the past year without any issues. I have not changed any settings in BIOS or so. It might be that Chrome OS automatically updated last night. 

**Any idea what I can do to get it booting or save my personal files?** 

- I managed to boot a live version of Ubuntu from a USB stick and can see my a drive where I believe my personal files are saved (though they seem to be somehow encrypted / inaccessible). 
- Is there a way to decrypt those files (with my Google login credentials)?

**Comment by tejasraman** [comment]
Hello @Vlad-BE 
Are you using ChromeOS flex or ChromeOS installed through brunch? Let me know and I can help you.

Tejas

**Comment by tejasraman** [comment]
I'm not 100 percent on encryption but I will look into it. I just saw your Reddit post so it's flex. 

This discussions page is technically for _brunch_ but I will help you with flex anyway. I'll look into recovering flex files with a Linux USB. (Weird that it encrypts stuff)

Tejas

**Reply by Vlad-BE** [reply]
Hi @tejasraman! Thanks a million! Indeed, I am using ChromeOS Flex. I used Brunch in the past but it also crashed. :( 
I have two other working laptops (with MacOS and Windows), USB stick etc - so would be incredibly grateful if you can let me know how to proceed. Thank you again!

**Reply by tejasraman** [reply]
Hi @Vlad-BE I need to get a VM set up to find a way to recover files so could you wait an hour or two? I promise I'll be back with a solution.

Tejas

**Reply by Vlad-BE** [reply]
Yes absolutely! I am based in Europe so still online for the next 5+ hrs. :) Thanks again!

**Reply by tejasraman** [reply]
As for brunch, I really think it can be troublesome (you need to update brunch before ChromeOS) but I use it anyway :)

Tejas

**Reply by Vlad-BE** [reply]
Yeah exactly - I was too hasty and excited about early updates before Brunch was updated... My bad!

**Reply by tejasraman** [reply]
It's ok, flex shouldn't crash either... weird that it did.

Maybe the bootloader is broken? If so we can fix it.

I will be back to you soon.

Tejas

**Reply by Vlad-BE** [reply]
I think the bootloader is broken. When I went to a 'legacy BIOS mode', I cannot see any OS in the launch list. Quite weird because I have not absolutely done anything other than surfing web. It's just disappeared. Hence I can only run live USB OS (Chrome OS or Ubuntu). Thanks again!

**Comment by tejasraman** [comment]
Hi! Could you go to the EFI mode?

Tejas
(P.S we probably need to repair the bootloader)

**Comment by tejasraman** [comment]
Sorry I couldn't make a VM for Flex but could you please boot into Ubuntu in EFI mode on your laptop? 
It is a thinkpad x260, right?

**Reply by Vlad-BE** [reply]
No worries. Yes I am back to "UEFI only" mode - and yes I can boot into Ubuntu Live USB.

**Comment by tejasraman** [comment]
Could you please run "sudo -s" and then type "cd /boot/efi/EFI/" and then "ls" and than take a screenshot?

**Comment by Vlad-BE** [comment]
See below:

![image](https://user-images.githubusercontent.com/116436933/197353179-4d13da8c-1b1c-4c17-ae53-b8fe87ac3bff.png)

**Comment by tejasraman** [comment]
Could you run `cd /boot && ls`?

**Comment by tejasraman** [comment]
Oh, and could you open "Disks" and take a screenshot for me?

**Comment by Vlad-BE** [comment]
Here you:
![image](https://user-images.githubusercontent.com/116436933/197353370-340842b8-2846-4127-a0ff-51e4b70c28a6.png)

![image](https://user-images.githubusercontent.com/116436933/197353384-b35a9d4d-e08f-4ae6-a25b-50a862cde7d3.png)

**Comment by tejasraman** [comment]
Can you click on "EFI-SYSTEM" and take another screenshot?

**Comment by tejasraman** [comment]
FYI Flex needs "EFI Mode" to work.

**Comment by Vlad-BE** [comment]
![image](https://user-images.githubusercontent.com/116436933/197353560-1b9e2aa4-cf2d-4b4a-8563-fc4029d6dcd2.png)

**Comment by tejasraman** [comment]
Could you run `udisksctl mount -b /dev/sda12`?

**Comment by Vlad-BE** [comment]
Yup - it says "mounted":
![image](https://user-images.githubusercontent.com/116436933/197353707-2c0fc788-6941-44d3-bb03-378513a9f3cf.png)

**Comment by tejasraman** [comment]
Can you run this: `umount /dev/sda12 && exit && udisksctl mount -b /dev/sda12`? Sorry, I made you mount it as root - that won't work. 

This unmounts it, exits "sudo -s", and mounts it again.

**Comment by Vlad-BE** [comment]
Thank you! This is what I did and can see: 
![image](https://user-images.githubusercontent.com/116436933/197353972-13f1ffbe-3ea7-4705-8fc2-d3a68e076e47.png)

**Comment by tejasraman** [comment]
Could you run `udisksctl mount -b /dev/sda12` again? I realized that "exit" in the above made it not mount.

Tejas

**Comment by Vlad-BE** [comment]
Fabulous - that worked: 
![image](https://user-images.githubusercontent.com/116436933/197354125-9a1e7737-c861-4a56-83c9-0f49c9a0e893.png)

**Comment by tejasraman** [comment]
Can you click "EFI-SYSTEM" in the Files sidebar?

**Comment by Vlad-BE** [comment]
Yup I can see this three folders
![image](https://user-images.githubusercontent.com/116436933/197354303-7f665873-683f-4c9f-bd09-57131419a601.png)

I went to EFI - Boot: 
![image](https://user-images.githubusercontent.com/116436933/197354336-9e9a355f-7870-49d5-8ce1-4a088e896d65.png)

**Comment by tejasraman** [comment]
So this is fixable.
Could you run this and reboot? 
`sudo efibootmgr --create --disk=/dev/sda --part=12 --label="ChromeOS Flex" --loader='efi\boot\grubx64.efi'`

**Comment by Vlad-BE** [comment]
Done - restarting. Wish me luck :))
![image](https://user-images.githubusercontent.com/116436933/197355148-61e79255-fd20-4d0b-8d7e-9a47200e527e.png)

**Comment by Vlad-BE** [comment]
Unfortunately still the same - Lenovo logo - black screen - Lenovo logo.

**Comment by Vlad-BE** [comment]
But I can see ChromeOS Flex now (which I could not see before):
![image](https://user-images.githubusercontent.com/116436933/197355277-5bfa8f62-01a9-4a95-8707-235b3cb0ed7f.png)

**Comment by tejasraman** [comment]
Does "ChromeOS Flex" boot?

**Comment by tejasraman** [comment]
If not, we need to run [Boot Repair](https://www.howtogeek.com/114884/how-to-repair-grub2-when-ubuntu-wont-boot/).....

**Comment by Vlad-BE** [comment]
Unfortunately it does not boot - have a look here:
https://photos.google.com/share/AF1QipMF_0stdmQF6HVRc0SQ5CmIFHfbskY9LtpELM4Czy7TrufiOb1A3gHt0mQxbYw76A/photo/AF1QipPOAGRMKMlfngV1NcaaiyKD_qVpnJdFfoEn6ws?key=bmFuaWFiWUl3T1o1aE9RVGNXcnVBMHh0V3lWanln

**Comment by Vlad-BE** [comment]
It is literally Lenovo logo - black screen - Lenovo logo in a loop.

**Comment by tejasraman** [comment]
Weird..... 
Could you try [Boot Repair](https://www.howtogeek.com/114884/how-to-repair-grub2-when-ubuntu-wont-boot/)?

**Comment by Vlad-BE** [comment]
Yes! Let me try - I will be back here in a few minutes. Thanks!!

**Comment by tejasraman** [comment]
Sorry for all the trouble and thanks so much!

Tejas

**Comment by Vlad-BE** [comment]
Hi Tejas, sorry for the delay! 
So it seems that the Boot Repair tool does not offer me the option of a "Recommended Repair" and it asks me to add the following TXT file straight away:
[Boot-Info_20221022_1851.txt](https://github.com/sebanc/brunch/files/9845092/Boot-Info_20221022_1851.txt)

![image](https://user-images.githubusercontent.com/116436933/197358059-541a022e-8130-40ec-a9c8-43bc21d8fe59.png)

Do you see anything extraordinary in that file? I am so sorry to be bothering you!!

**Comment by Vlad-BE** [comment]
![image](https://user-images.githubusercontent.com/116436933/197358142-eb9b25ed-2bf6-4dc8-8f73-ed4f32417eef.png)

**Comment by tejasraman** [comment]
I'm sorry, your entire boot loader is corrupt.

I'm not sure how to fix it either. Reinstalling GRUB won't really work for me.

I'm sorry that I could not be of much help to you. Do you have the info on the drive backed up?

![image](https://user-images.githubusercontent.com/96961804/197358195-4c8ce7e9-a210-47b1-834a-50351e7c7e44.jpeg)

**Comment by Vlad-BE** [comment]
Ah pity! Do not worry! I do regular monthly backups but unfortunately I have lost around 1.5 month of personal data (including family pics from our holiday which is particularly painful). I have already deleted stuff from G Drive and G Photos and was about to do the backup of the drive. :(

I assume that I cannot just use a GRUB data from another working Chrome OS installation, right?

A million thanks for all your help and support!!

**Comment by Vlad-BE** [comment]
Perhaps an additional question: Is there a way to decrypt the partition that includes my personal data? I have come across [this](https://www.chromium.org/developers/design-documents/encrypted-partition-recovery/) article but I wonder whether this is possible (i.e. using my google login credentials)?

**Comment by tejasraman** [comment]
You might be able to use grub from another flex install. I will send it to you.

Tejas

**Comment by tejasraman** [comment]
Could you send me "/efi/boot/grub.cfg" ?

I have a working Flex EFI and we can fix your EFI but I need that file so I can send you new files.

Tejas

**Comment by Vlad-BE** [comment]
Hi Tejas! A million thanks! 
See the file here: 
[efi.zip](https://github.com/sebanc/brunch/files/9845959/efi.zip)

Hope it works!!

**Comment by heitre** [comment]
@Vlad-BE Prepare an USB with  Super Grub2 Disk
 https://www.supergrubdisk.org/ . Boot it. It will give you a list of all kernels; Choose  some, try.

**Comment by Vlad-BE** [comment]
Interesting tool @heitre - I keep getting "No grub.cfg file detected" whenever I try to "detect and show boot methods" or "boot manually" with the help of the tool. Hence I believe the step that @tejasraman proposed (meaning replacing the CFG file) might help?

**Comment by heitre** [comment]
Do you mean it does not list any kernels when you  "detect and show boot methods"?

**Comment by Vlad-BE** [comment]
Let me post several pics from the process after I click on "detect and show boot methods":

![image](https://user-images.githubusercontent.com/116436933/197388438-3feba085-dea2-4fc6-a3a7-45e85595b4fe.png)
![image](https://user-images.githubusercontent.com/116436933/197388444-718b3e5d-db84-445c-955a-a746f8261c80.png)
![image](https://user-images.githubusercontent.com/116436933/197388451-16fe833c-dbbd-4c97-bd17-1daab58483a3.png)
![image](https://user-images.githubusercontent.com/116436933/197388461-bb64b81c-0979-4ab7-8e3f-b3203c50d29c.png)

Would you be so kind and let me know what option I should click on in the "orange" screen?

**Comment by heitre** [comment]
try starting from 1.. may be this works already. If not try 2...
![1111](https://user-images.githubusercontent.com/107424361/197389652-01f9c9d0-f6b0-4259-96b2-2bc137ac90db.png)

(If something does NOT work. Turn off restart again)

**Comment by Vlad-BE** [comment]
Thanks a lot - I tried all six options. After I click on any of them, the screen stays black for like 5 seconds and then the computer restarts back to the Lenovo welcome logo and the USB stick loads the "Super GRUB2 disk" tool. :( 
Should I try other options in this tool?

**Comment by heitre** [comment]
How old is this X260? I suspect the HDD/SSD is may be giving up... One possibility is that you get another SSD, boot to ubuntu, clone/dd the entire disk elsewhere.

It just could be that the disk failed during updates... (i.e) Not a total fail but with too many bad sectors. 

PS: May be try booting with ubuntu and then locate the partition containing` vmlinuz (gpt5 or gpt3)` and try 

   ` fsck.ext4 -p /dev/SOMETHINGHERE`

Edit: Did it work?

**Reply by Vlad-BE** [reply]
Apologies - just to make sure I am doing it correctly. So I should run this: 
"sudo fsck -p /dev/sda3"
and also 
"sudo fsck -p /dev/sda5"

Is that correct please? As per some online instructions, I understand I have to unmount the disks. 

Thank you @heitre!

EDIT: And yes, the laptop is from 2016 if I remember correctly. But the SSD is from 2017 or 2018. The original laptop came with an old-school HDD which I replaced with a faster SSD.

**Comment by heitre** [comment]
Yes, try that. And then 

1. restart it. Hope it automatically works. if not

2. Use  Super Grub2 Disk again... and select things as shown in the screenshot

**Comment by tejasraman** [comment]
You know @heitre fsck is a good suggestion. My main problem is that the EFI seems to have bad AVD sectors.
I'm wondering if he should wipe the EFI and put new files (just as a test).

**Comment by tejasraman** [comment]
Although after what happened with the grub2 tool I wonder how deep the corruption goes.

My question is if we could somehow decrypt the data... Maybe with a chromium/Chrome OS USB?

**Comment by heitre** [comment]
Edit: Don't do this

Given I do **not** understand the boot procedure I would not wipe. May be` fsck.vfat -a  /dev/SOMETHINGHERE ` on the EFI partition?

**Reply by tejasraman** [reply]
@heitre NEVER FSCK AN EFI PARTITION UNLESS YOU WANT TO BREAK IT. Seriously.

**Reply by tejasraman** [reply]
Let's see.... Grub: broken. Kernels: broken. Data: Google encrypts it.


Ugh.

**Reply by heitre** [reply]
Sorry. I assumed EFI is just FAT32 partition. Isn't it?

**Reply by tejasraman** [reply]
For him it's fat16 (and probably dualFAT too)

**Comment by tejasraman** [comment]
Hi @Vlad-BE
Obviously something deeper than the bootloader. If fsck doesn't work maybe the data can be decrypted.

I know the method on Chrome OS devices but flex devices are different.

Could you find the path of your encrypted files, install fscrypt, and unlock it?
`fscrypt unlock yourcrosinstall`

**Reply by heitre** [reply]
`fscrypt unlock yourcrosinstall
`

But what would be the password for this? Surely, Chromeos should use some other key to decrypt rather than simple google password. What do you think?

**Reply by tejasraman** [reply]
I mean if you use a real ChromeOS device your google password is the password (or some combination of the email and password). I'm just curious though so it might not work.

P.S you can set it to use a pass phrase in settings

**Reply by heitre** [reply]
> or some combination of the email and password)


How is the user supposed to know what to type?

> P.S you can set it to use a pass phrase in settings

Now?

**Reply by tejasraman** [reply]
I'm not sure. And not now, but when it was bootable.

**Comment by Vlad-BE** [comment]
Hi both @tejasraman & @heitre! Sorry for the delayed response - had to take care of kids.. 

So I tried to run "sudo fsck -p /dev/sda3" and "sudo fsck -p /dev/sda5" beg this screen:

![image](https://user-images.githubusercontent.com/116436933/197400565-58a73854-3267-4586-b808-9bf40abe02ff.png)

Is there a better command for this, or I assume I have to install that software? 

I can try to decrypting those files in parallel. In fact, this might be an easier option as I know which files I have not backed up and I would simply format the drive afterwards or buy a new SSD.

**Reply by tejasraman** [reply]
No.... ChromeOS is using some unsupported stuff. Could you try the fscrypt command?

**Comment by heitre** [comment]
Try using newer Ubuntu ISO/USB.

> `fsck`

This does not do any decryption. It is just a check for filesystem. My hope is that the `data partition` is safe and the `system or EFI` is broken. so fixing the system partition. Then somehow boot it successfully. Then chromeos will do the automatic decryption.

Sadly, I guess is that SSD somehow failed. This would mean also data may be lost. With encrypted partitions - from my understanding -  errors in SSD would render the disk useless. That is the design.

**Comment by tejasraman** [comment]
You know @heitre you could be right. The last time I suspected SSD failure though it ended up needing an OS reinstall. (Some apps that relied on a particular windows mount point didn't work)

**Comment by tejasraman** [comment]
FSCK is not for decryption. FSCrypt is.

As for SSDs, the SK Hynix Gold S31 and Adata SU800 are really reliable in my experience.

**Comment by Vlad-BE** [comment]
Running the 'check disk' command worked for the partition where I believe my personal data is saved.  Hopefully this means that this partition is safe?

![image](https://user-images.githubusercontent.com/116436933/197405297-db361c2a-5330-42bb-9897-93654e60ceb5.png)

In terms of FSCrypt, would you be so kind and help me install this tool? I have been following some instructions online but get this error:

![image](https://user-images.githubusercontent.com/116436933/197405695-fa43ca3d-ff1c-4cd0-8d8a-139cf02e2a0c.png)

EDIT: 
I have also scanned the SSD for failures in Lenovo's boot diagnostics option. After a 5 min scan, everything was "green" (passed successfully).

**Reply by tejasraman** [reply]
Run `sudo apt update` before the install command.

**Reply by Vlad-BE** [reply]
Ah true! Still learning - sorry! Unfortunately, it tells me this: 
![image](https://user-images.githubusercontent.com/116436933/197406933-8a61e5ca-7c06-444c-92f7-23c1e1d53b32.png)

Does it mean that my Ubuntu 22.10 is too new? Perhaps I may have to run a previous version? It was the same issue for me with the Boot Repair tool we tried yesterday.

**Reply by heitre** [reply]
```
sudo apt-get update 
sudo apt-cache search -n fscrypt
```

To be very honest, I would get the hopes down as chromeos is very security first one. 

Not sure you will ever decrypt it.

**Reply by Vlad-BE** [reply]
Thanks @heitre & @tejasraman! So I had to download and install the Fscrypt tool manually. Now I am not sure how to run it. 

I suspect that my files are here:
![image](https://user-images.githubusercontent.com/116436933/197409868-7eef1d74-7fd5-426d-913b-e9d14272d6e6.png)

This is what I see when I open that partition:
![image](https://user-images.githubusercontent.com/116436933/197409843-e3bca5f9-2ba2-48e1-a1ee-82e86398fac0.png)

Would you be so kind and guide me to unlock the files please? 

I cannot thank you enough!

**Reply by heitre** [reply]
`fscrypt status 
`
see the status command here https://wiki.archlinux.org/title/Fscrypt

(Pretty sure chromeos is using some encrypted.key - see the screenshot). We are just randomly issuing commands without understanding). Sorry.

> As for SSDs, the SK Hynix Gold S31 and Adata SU800 are really reliable in my experience.

Ask in reddit datahoarder regarding any disk/ssd/hdd - The answer for reliability is a number between 1 sec to any large number.

**Reply by Vlad-BE** [reply]
Thanks @heitre ! Btw @tejasraman mentioned that I may need a [eCryptfs](https://ecryptfs.org/) tool from Chrome OS files - do you have any experience with it?

**Reply by heitre** [reply]
eCryptfs is easy. We all know/can learn commandline. But we do not know how it is implemented inside chromeos. We do not know which password? Most  likely it is not your direct password; it should be some KEY file. May be this is also corrupted. see your screenshot.

Sorry. It is over. Nothing can be done.

**Reply by tejasraman** [reply]
Ya that's the exact reason I didn't want to meddle with eCryptFS. The thing is that fscrypt is a Google-developed utility and they use it in ChromeOS. Real Chromebooks use a TPM-stored Google account "protector" to decrypt the files.

Google might have messed up eCryptFS and we don't know. I'm so sorry @Vlad-BE but if the files are in GDrive you can get them.

Also your quote "everyone knows/can learn command line" is very true :) @heitre

**Comment by tejasraman** [comment]
What id inside unencrypted?

And try "fscrypt unlock yourencrypteddir"

**Reply by Vlad-BE** [reply]
The unencrypted folder contains this: 
![image](https://user-images.githubusercontent.com/116436933/197413032-210e67ae-302d-47ca-b43e-7cd448273aef.png)

So I write this for instance? 
fscrypt unlock /dev/sda1
or
fscrypt unlock /media/ubuntu/d5ede90c-273a-49d0-a5c7-2638e67a3cc4

**Comment by tejasraman** [comment]
The second one.

**Reply by Vlad-BE** [reply]
This is what I get - am I doing it correctly?
![image](https://user-images.githubusercontent.com/116436933/197414187-91fc9951-33a7-4ac2-b2c0-bdf59de16331.png)

**Comment by tejasraman** [comment]
Maybe /encrypted ?

**Reply by Vlad-BE** [reply]
Looks same to me: 
![image](https://user-images.githubusercontent.com/116436933/197414365-068bedac-c672-429f-b3d8-2ac682ee796a.png)

Weirdly, the encrypted folder looks empty. 
![image](https://user-images.githubusercontent.com/116436933/197414391-04bf000f-ea92-4770-8d71-583e7ca01bd5.png)

**Comment by tejasraman** [comment]
What about /home/chronos ? (yes it is C-H-R-O-N-O-S)

**Reply by Vlad-BE** [reply]
No luck my friend :( 
![image](https://user-images.githubusercontent.com/116436933/197418074-9da9b019-5737-4811-8975-dfadfac4843e.png)

**Reply by Vlad-BE** [reply]
@heitre Maybe you know?

**Reply by tejasraman** [reply]
Ugh I meant inside the media folder (/media/Ubuntu/_whatever_/home/chronos)

**Reply by Vlad-BE** [reply]
Ah got it! Sorry! I tried a couple of options but either it says that the directory is not encrypted, or that I do not have sufficient permissions (as if I was not the owner). 
![image](https://user-images.githubusercontent.com/116436933/197418832-aa4fcce9-87c5-41d3-b489-3c8a8e3ed8c3.png)

**Reply by tejasraman** [reply]
Add sudo the beginning?

**Reply by Vlad-BE** [reply]
Apparently "not encrypted": 
![image](https://user-images.githubusercontent.com/116436933/197418934-fd696b7b-40b3-4de2-9f32-71da7d5a9aee.png)

**Reply by tejasraman** [reply]
What is inside that home directory?

**Reply by Vlad-BE** [reply]
![image](https://user-images.githubusercontent.com/116436933/197419007-8dc8b0cf-ba80-402f-b040-6a0c69191952.png)

**Comment by tejasraman** [comment]
Try "chronos"

**Reply by Vlad-BE** [reply]
It looks empty:
![image](https://user-images.githubusercontent.com/116436933/197419097-747839e2-0058-44e6-a919-65f4ae518831.png)

And here with commands (sudo and non sudo):
![image](https://user-images.githubusercontent.com/116436933/197419168-6109d0d6-9105-479d-b6f9-95c2085126dd.png)

**Comment by tejasraman** [comment]
It might be eCryptFS as well (Some older chromebooks used it even in modern ChromeOS versions)

![image](https://user-images.githubusercontent.com/96961804/197419101-058b5ae5-70bc-4972-a34e-bac91084f21d.jpeg)

**Comment by tejasraman** [comment]
If so, there's not much we can do about the data. I know your ssd is fine but I wonder how the OS got corrupted.

**Reply by Vlad-BE** [reply]
Got it! Yesterday, you proposed to load a format and load a new EFI? Do you think this could still work please? @tejasraman

**Reply by tejasraman** [reply]
I'll try. 
I'll send you a new EFI.
Can you
- Reformat the Efi partition as fat16
- Add the Efi files to that partition?

**Reply by Vlad-BE** [reply]
Uff, with your help for sure - but how much more can I bother you...? :D

**Reply by tejasraman** [reply]
True :-) but it's pretty low effort to replace the EFI.

**Reply by Vlad-BE** [reply]
I cannot thank you enough. I would be grateful if you can send me even a link to instructions for the EFI reformatting / loading new files (or write it here if you have a moment and patience with me). Thanks!

**Reply by Vlad-BE** [reply]
@tejasraman  - It seems to me that I already have EFI partition as fat16?
![image](https://user-images.githubusercontent.com/116436933/197419564-82497e5d-f473-4b22-b267-4a08cb100aab.png)

So I would just need to replace / load new files?

**Comment by tejasraman** [comment]
It’s probably eCryptFS then.

**Reply by Vlad-BE** [reply]
And I cannot use similar tool like fscrypt to decrypt eCryptFS?

**Comment by tejasraman** [comment]
Can you run these commands in order:
`sudo apt update`
`sudo apt install dosfstools`
`sudo mkfs.vfat /dev/sda12`
`sudo e2label /dev/sda12 "EFI"`
`udisksctl mount /dev/sda12`

And then put the contents of https://1drv.ms/u/s!AkBuD6XsET8xgYwJHBXY7H_kmPc-Pw on the new partition.

**Comment by tejasraman** [comment]
This erases the partition (just to make sure) and labels it as "EFI"

**Reply by Vlad-BE** [reply]
have a look - I made it to the last step before copying the files. It seems that the EFI label has not been applied:
![image](https://user-images.githubusercontent.com/116436933/197419804-31536884-5945-4b26-9bee-ec9c72d3f522.png)

![image](https://user-images.githubusercontent.com/116436933/197419816-1e373c7c-a398-41f7-a27e-94e21543a8bf.png)

**Reply by Vlad-BE** [reply]
@tejasraman  But I guess just copy those files from You in the EFI partition, even if the label may not have changed?

**Reply by Vlad-BE** [reply]
Weirdly, in the Disks app, it looks normal? Maybe I can change the name here?
![image](https://user-images.githubusercontent.com/116436933/197420153-33ed34a1-a60c-4487-ab4d-a59f6f6cb32b.png)

![image](https://user-images.githubusercontent.com/116436933/197420504-08246926-ec5b-49db-a49f-9f5c1122bc5f.png)

**Reply by tejasraman** [reply]
You can. Remember to use udisksctl on it and transfer the files.

**Comment by Vlad-BE** [comment]
Thanks @tejasraman! So I copied your new EFI files into the formatted 'EFI' partition. Unfortunately, that did not help: the PC is still in a loop: Lenovo welcome screen - black screen - Lenovo welcome screen... :( 

Do you have any more ideas please? I basically need to save the get some data from the SSD disk. 

1. I wonder if I should try the eCryptFS decryption method too? 
2. Or can something be changed in the BIOS settings to make this thing boot (i.e. secure boot, TMP chip or so)?

**Reply by tejasraman** [reply]
I'm not sure. Maybe try boot repairing the new EFI?

Just to get a log and see if there are changes and maybe repair it?

**Reply by Vlad-BE** [reply]
@tejasraman See here the Boot Repair report - what do you think?
[Boot-Info_20221024_1655.txt](https://github.com/sebanc/brunch/files/9853699/Boot-Info_20221024_1655.txt)

**Comment by Vlad-BE** [comment]
Hi @tejasraman, I guess there is not much I can do with this Boot Repair log, right? What do you think? I want to make sure it is the right time to just give it up and format the disk.
[Boot Repair Info.txt](https://github.com/sebanc/brunch/files/9853699/Boot-Info_20221024_1655.txt)
Thanks again for everything!

**Reply by tejasraman** [reply]
It still doesn't boot? The EFI is fine now (the log says so) so the file corruption runs deeper than we can fix.

![Screenshot_20221025-135121.png](https://user-images.githubusercontent.com/96961804/197868687-f8180d83-b1fb-4d75-b6dc-7b84687c2981.png)

IMHO just give up.

**Reply by Vlad-BE** [reply]
Thanks @tejasraman - totally understand. I really appreciate all your help and support throughout this journey..!

**Reply by tejasraman** [reply]
I'm so sorry I couldn't help...
